lib/puppet-strings/yard/tags/property_directive.rb

```diff
@@ -0,0 +1,24 @@
+require 'puppet-strings/yard/code_objects'
+
+# Implements a parameter directive (e.g. #@!puppet.type.param) for documenting Puppet resource types.
+class PuppetStrings::Yard::Tags::PropertyDirective < YARD::Tags::Directive
+  # Called to invoke the directive.
+  # @return [void]
+  def call
+    return unless object && object.respond_to?(:add_property)
+    # Add a property to the resource
+    property = PuppetStrings::Yard::CodeObjects::Type::Parameter.new(tag.name, tag.text)
```

Comment:
Perhaps we should subclass a `Property` class from `Parameter` that has no implementation just so this isn't as confusing.